src/index.c

```diff
@@ -144,7 +144,7 @@ IndexIterator *NewUnionIterator(IndexIterator **its, int num, DocTable *dt, int
     }
   }
 
-  const size_t maxresultsSorted = 1000;
+  const size_t maxresultsSorted = RSGlobalConfig.maxResultsToUnsortedMode;
   if (it->mode == MODE_SORTED && ctx->nexpected >= maxresultsSorted) {
```

Comment:
Please add a comment here describing why this code is normally (and should be) dead. i.e. the deepest-most IndexIterator does not have a tester